File: backend/app/schemas/notification.py
"""Schemas for in-app notifications."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import NotificationType


class NotificationRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    recipient_id: uuid.UUID
    institution_id: uuid.UUID
    notification_type: NotificationType
    title: str
    body: str
    is_read: bool
    audit_id: uuid.UUID | None
    created_at: datetime

File: frontend/src/types/notification.ts
export type NotificationType =
  | "audit_assigned"
  | "due_soon"
  | "overdue"
  | "evidence_uploaded"
  | "evidence_missing"
  | "audit_returned"
  | "audit_approved"
  | "audit_rejected"
  | "audit_completed"
  | "new_comment";

export interface Notification {
  id: string;
  recipient_id: string;
  institution_id: string;
  notification_type: NotificationType;
  title: string;
  body: string;
  is_read: boolean;
  audit_id: string | null;
  created_at: string;
}
